packages/host/app/components/operator-mode/container.gts

```diff
@@ -367,15 +363,7 @@ export default class OperatorModeContainer extends Component<Signature> {
         stackIndex: 0,
       };
 
-      let currentStackItems =
-        this.operatorModeStateService.state.stacks[0].items;
-
-      currentStackItems.forEach((item) => {
-        this.operatorModeStateService.replaceItemInStack(item, {
-          ...item,
-          stackIndex: 1,
-        });
-      });
```

Comment:
This is a bug here I found 

It is a bad idea to mutate `this.state.stacks[0].items` when looping thru it. Particularly, if we are incrementing index. I believe if we reversed the index or use a pop-style pattern with this `forEach` it would work. 

This seems to be an issue if you use `removeItemsFromStack` with `forEach`. It is not an issue when `addItemsToStack` with`forEach`. My solution was to introduce a new function `removeItemsIf`